src/relax/op/ccl/ccl.cc

```diff
@@ -50,6 +50,44 @@ TVM_REGISTER_OP("relax.ccl.allreduce")
     .set_attr<FRelaxInferLayout>("FRelaxInferLayout", InferLayoutUnaryEwise)
     .set_attr<Bool>("FPurity", Bool(true));
 
+/* relax.ccl.allgather */
+TVM_REGISTER_NODE_TYPE(AllGatherAttrs);
+
+Expr allgather(Expr x, int num_workers) {
+  ObjectPtr<AllGatherAttrs> attrs = make_object<AllGatherAttrs>();
+  attrs->num_workers = std::move(num_workers);
+  static const Op& op = Op::Get("relax.ccl.allgather");
+  return Call(op, {std::move(x)}, Attrs{attrs}, {});
+}
+
+TVM_REGISTER_GLOBAL("relax.op.ccl.allgather").set_body_typed(allgather);
+
+StructInfo InferStructInfoAllGather(const Call& call, const BlockBuilder& ctx) {
+  TensorStructInfo input_sinfo = GetUnaryInputTensorStructInfo(call, ctx);
+  DataType output_dtype = input_sinfo->dtype;
+
+  const auto* attrs = call->attrs.as<AllGatherAttrs>();
+  int num_workers = attrs->num_workers;
+  auto input_shape = input_sinfo->GetShape();
+  if (!input_shape.defined()) {
+    return input_sinfo;
+  }
+  Array<PrimExpr> output_shape = input_shape.value();
+  output_shape.Set(0, floor(output_shape[0] * num_workers));
+  if (input_sinfo->vdevice.defined()) {
```

Comment:
Instead of duplicating the entire `return TensorStructInfo(...)` call, we can make a local variable.  Otherwise, readers would need to inspect the 

```c++
VDevice vdevice;
if(input_sinfo->vdevice.defined()) {
    vdevice = input_sinfo->vdevice.value();
}
return TensorStructInfo(ShapeExpr(output_shape), output_dtype, vdevice);
```